src/Workspaces/Core/Portable/LinkedFileDiffMerging/LinkedFileDiffMergingAggregatedLogger.cs

```diff
@@ -0,0 +1,200 @@
+        }
+    }
+
+    internal class LinkedFileDiffMergingSessionInfo
+    {
+        public readonly List<LinkedFileGroupSessionInfo> LinkedFileGroups = new List<LinkedFileGroupSessionInfo>();
+
+        public void LogLinkedFileResult(LinkedFileGroupSessionInfo info)
+        {
+            LinkedFileGroups.Add(info);
+        }
+    }
+
+    internal class LinkedFileGroupSessionInfo
+    {
+        public int LinkedDocuments;
+        public int DocumentsWithChanges;
+        public int IsolatedDiffs;
+        public int IdenticalDiffs;
+        public int OverlappingDistinctDiffs;
+        public int OverlappingDistinctDiffsWithSameSpan;
+        public int OverlappingDistinctDiffsWithSameSpanAndSubstringRelation;
+        public int InsertedMergeConflictComments;
+        public int InsertedMergeConflictCommentsAtAdjustedLocation;
+    }
+
+    internal static class SessionLogMessage
+    {
+        private const string SessionId = "SessionId";
+
+        private const string LinkedDocuments = "LinkedDocuments";
+        private const string DocumentsWithChanges = "DocumentsWithChanges";
+        private const string IdenticalDiffs = "IdenticalDiffs";
+        private const string IsolatedDiffs = "IsolatedDiffs";
+        private const string OverlappingDistinctDiffs = "OverlappingDistinctDiffs";
+        private const string OverlappingDistinctDiffsWithSameSpan = "OverlappingDistinctDiffsWithSameSpan";
+        private const string OverlappingDistinctDiffsWithSameSpanAndSubstringRelation = "OverlappingDistinctDiffsWithSameSpanAndSubstringRelation";
+        private const string InsertedMergeConflictComments = "InsertedMergeConflictComments";
+        private const string InsertedMergeConflictCommentsAtAdjustedLocation = "InsertedMergeConflictCommentsAtAdjustedLocation";
+
+        private const string TotalSessions = "TotalSessions";
+        private const string TotalSessionsWithLinkedFiles = "TotalSessionsWithLinkedFiles";
+        private const string TotalLinkedFileGroupsProcessed = "TotalLinkedFileGroupsProcessed";
+        private const string TotalIdenticalDiffs = "TotalIdenticalDiffs";
+        private const string TotalIsolatedDiffs = "TotalIsolatedDiffs";
+        private const string TotalOverlappingDistinctDiffs = "TotalOverlappingDistinctDiffs";
+        private const string TotalOverlappingDistinctDiffsWithSameSpan = "TotalOverlappingDistinctDiffsWithSameSpan";
+        private const string TotalOverlappingDistinctDiffsWithSameSpanAndSubstringRelation = "TotalOverlappingDistinctDiffsWithSameSpanAndSubstringRelation";
+        private const string TotalInsertedMergeConflictComments = "TotalInsertedMergeConflictComments";
+        private const string TotalInsertedMergeConflictCommentsAtAdjustedLocation = "TotalInsertedMergeConflictCommentsAtAdjustedLocation";
+
+        public static KeyValueLogMessage Create(int sessionId, LinkedFileDiffMergingAggregatedLogger aggregateInfo)
```

Comment:
This is a new structure with new keys, etc. Do I need to define this externally, or does the KeyValueLogMessage make everything just work on the server side?